usecases/schema/property.go

```diff
@@ -16,47 +16,92 @@ import (
 	"fmt"
 	"strings"
 
-	"github.com/pkg/errors"
 	"github.com/weaviate/weaviate/entities/models"
 	"github.com/weaviate/weaviate/entities/schema"
 )
 
-var errPropertyNotFound = errors.New("property not found")
-
-// AddClassProperty to an existing Class
+// AddClassProperty it is upsert. it adds properties to a class and updates
+// existing properties if the merge bool passed true.
 func (h *Handler) AddClassProperty(ctx context.Context, principal *models.Principal,
-	class string, prop *models.Property,
+	class *models.Class, merge bool, newProps ...*models.Property,
 ) error {
+	if len(newProps) == 0 {
+		return nil
+	}
+
 	err := h.Authorizer.Authorize(principal, "update", "schema/objects")
 	if err != nil {
 		return err
 	}
 
-	if prop.Name == "" {
-		return fmt.Errorf("property must contain name")
-	}
-	prop.Name = schema.LowercaseFirstLetter(prop.Name)
-	if prop.DataType == nil {
-		return fmt.Errorf("property must contain dataType")
+	// validate new props
+	for _, prop := range newProps {
+		if prop.Name == "" {
+			return fmt.Errorf("property must contain name")
+		}
+		prop.Name = schema.LowercaseFirstLetter(prop.Name)
+		if prop.DataType == nil {
+			return fmt.Errorf("property must contain dataType")
+		}
 	}
 
-	cls := h.metaReader.ReadOnlyClass(class)
-	if cls == nil {
-		return fmt.Errorf("class %q: %w", class, ErrNotFound)
+	existingNames := map[string]bool{}
```

Comment:
todo: we can move the instantiation of the map after the `setNewPropDefaults` call and use `make` to initialize it's capacity based on the length of `class.Properties` if `!merge` to avoid many small mem alloc.